src/main.rs

```diff
@@ -99,86 +102,160 @@ struct Args {
     connector: Option<String>,
 }
 
-fn main() {
-    let args = Args::parse();
+struct DrmDisplay {
+    gbm_device: GbmDevice<Card>,
+    crtc: CrtcInfo,
+    connector: ConnectorInfo,
+    mode: Mode,
+    width: u32,
+    height: u32,
+}
 
-    // TODO(bschwind) - Use libdrm to iterate over available DRM devices.
-    let gpu = Card::open(&args.card_path);
-    dbg!(gpu.get_driver().expect("Failed to get GPU driver info"));
-    dbg!(gpu.get_bus_id().expect("Failed to get GPU bus ID"));
+impl DrmDisplay {
+    fn init(args: &Args) -> Option<DrmDisplay> {
```

Comment:
This feels like a `new` method, so let's call it that by common convention. Also looking at the ways it returns `None`, it feels like this should return a `Result` instead.

Reply:
True, I'll rename it.

Using `Result` doesn't really help until we start handling errors properly. I did not really touch the DRM init code here, so it should be possible to work on #6 in parallel with this and then merge.